### PR TITLE
[feat] Homepage

### PR DESCRIPTION
**Pull request by shivampandey0**
- Added DataContext for videos 
- getvideos from backend 
- VideoCard component for Homepage with WatchLater CTA
- Homepage responsiveness
- Categories on Homepage
- Chip component for Category

> **Note:** Need to show some error when there's no data from API or no videos in the selected category, will be done later by raising the issue

> Categories are hard-coded now, will be refactored with backend route when implemented

image

### CODE REVIEW

File: src/utils/server-request.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { ACTION_TYPES, requests } from './constants';
+import { axios } from './index';
+
+const getVideos = async (dispatch) => {
+  try {
+    const res = await axios.get(requests.videos);
+    console.log(res);
```
Comment (saadahmadhere) [inline src/utils/server-request.js]: remove unwanted console.log

**Review by saadahmadhere — APPROVED** [review]
- Nicely written code and beautiful file structure.
- Good to see the use of constant variables and the way it is stored in an object, and used. 
- Looking forward to learn from you how to you implemented your own backend.

File: src/utils/server-request.js (hunk 1 above)
Comment (shivampandey0) [reply src/utils/server-request.js]: Done.

**Review by GuptaEesh — COMMENTED** [review]
- Good use of skeleton loader.
- Good use of try-catch block
- Good use of barrel imports.

Good to merge

**Pull request merged**